server/container/util/src/test/java/org/apache/james/util/concurrency/ConcurrentTestRunnerTest.java

```diff
@@ -132,4 +133,79 @@ public void consume(int threadNumber, int step) throws Exception {
         assertThat(concurrentTestRunner.awaitTermination(DEFAULT_AWAIT_TIME, TimeUnit.MILLISECONDS)).isTrue();
         assertThat(queue).containsOnly("0:0", "0:1", "1:0", "1:1");
     }
+
+    @Test
+    public void runShouldNotThrowOnExceptions() throws Exception {
+        int operationCount = 2;
+        int threadCount = 2;
+
```

Comment:
I feel stupid and I do not understand why this is working just couple line before you have:
```
+            if (exception != null) {
+                throw Throwables.propagate(exception);
+            }
```
so what I understand is that run is actually throwing error. I guess I am missing something but what..